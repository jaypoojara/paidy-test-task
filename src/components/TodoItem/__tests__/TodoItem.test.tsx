import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import ToDoItem from "../index";

describe("ToDoItem Component", () => {
  const mockOnPressTodo = jest.fn();
  const mockOnPressRemove = jest.fn();
  const title = "Test Todo";
  const index = 1;

  it("renders correctly", () => {
    const element = render(
      <ToDoItem
        onPressTodo={mockOnPressTodo}
        onPressRemove={mockOnPressRemove}
        title={title}
        index={index}
      />
    );
    expect(element).toMatchSnapshot();
  });

  it("renders correctly with the provided props", () => {
    const { getByText, getByTestId } = render(
      <ToDoItem
        onPressTodo={mockOnPressTodo}
        onPressRemove={mockOnPressRemove}
        title={title}
        index={index}
      />
    );

    const todoContainer = getByTestId("todoItemTestID");
    const todoTitle = getByText(title);

    expect(todoContainer).toBeDefined();
    expect(todoTitle).toBeDefined();

    fireEvent.press(todoContainer);
    expect(mockOnPressTodo).toHaveBeenCalledWith(index);

    const removeButton = getByText("REMOVE");
    fireEvent.press(removeButton);
    expect(mockOnPressRemove).toHaveBeenCalledWith(index);
  });
});
