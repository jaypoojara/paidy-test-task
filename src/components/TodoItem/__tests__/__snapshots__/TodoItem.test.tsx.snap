// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoItem Component renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 15,
      "flexDirection": "row",
      "marginVertical": 5,
      "opacity": 1,
      "padding": 15,
    }
  }
  testID="todoItemTestID"
>
  <View
    style={
      {
        "backgroundColor": "#034096",
        "borderRadius": 10,
        "height": 20,
        "width": 20,
      }
    }
  />
  <Text
    style={
      {
        "color": "#7a7a7a",
        "flex": 1,
        "fontSize": 15,
        "fontWeight": "500",
        "paddingHorizontal": 10,
      }
    }
  >
    Test Todo
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#7a7a7a",
          "fontWeight": "500",
        }
      }
    >
      REMOVE
    </Text>
  </View>
</View>
`;

exports[`TodoItem Component renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 15,
      "flexDirection": "row",
      "marginVertical": 5,
      "opacity": 1,
      "padding": 15,
    }
  }
  testID="todoItemTestID"
>
  <View
    style={
      {
        "backgroundColor": "#034096",
        "borderRadius": 10,
        "height": 20,
        "width": 20,
      }
    }
  />
  <Text
    style={
      {
        "color": "#7a7a7a",
        "flex": 1,
        "fontSize": 15,
        "fontWeight": "500",
        "paddingHorizontal": 10,
      }
    }
  >
    Test Todo
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#7a7a7a",
          "fontWeight": "500",
        }
      }
    >
      REMOVE
    </Text>
  </View>
</View>
`;
